Node/core/src/bots/ChatConnector.ts

```diff
@@ -181,95 +191,131 @@ export class ChatConnector implements IConnector, IBotStorage {
             var verifyOptions: jwt.VerifyOptions;
             var openIdMetadata: OpenIdMetadata;
             const algorithms: string[] = ['RS256', 'RS384', 'RS512'];
+            
+            if(this.settings.enableSkills && SkillValidation.isSkillToken(authHeaderValue)) {
+                const skillMsg = utils.clone(req.body);
```

Comment:
utils.clone isn't a deep clone

If you need a deep clone, the simplest way is to use `JSON.parse(JSON.stringify(req.body))`, though it is inefficient. There are other deep clone implementations on the interwebs you could copy.

Reply:
not sure if that was intentional or not (but i'm guessing not) as we pulled this code in from v4 - but used 'deep-clone' npm module to perform deep clone of req.body.